js/the_hashtag_generator.js

```diff
@@ -0,0 +1,18 @@
+// https://www.codewars.com/kata/the-hashtag-generator/train/javascript
+
+function generateHashtag(str) {
+  String.prototype.capitalize = function() {
+    return this.replace(/(?:^|\s)\S/g, function(a) {
+      return a.toUpperCase();
+    });
+  };
+
+  const strWithoutSpaces = str.capitalize().replace(/\s/g, '');
```

Comment:
Consider splitting the string by any whitespace, and then mapping over the results. Follow it up with joining the results and prepending the `#` 😄 the solution would most likely be more straightforward.

Regexps such as the one used in `capitalize` are harder to read and take longer to understand than regular functions 🙂